### PR TITLE
Fox bugs

### PR DESCRIPTION
**Pull request by FelixJacobsen**
This will fix the bugs we had with creating an account with same username multiple times, add a respond if trying to login with username that doesn't exist and change authenticate token to a arrow function.

### CODE REVIEW

**Comment by LordRekishi** [comment]
[Account: Not add same username twice](https://trello.com/c/pg4aqGht/204-account-not-add-same-username-twice)

**Comment by LordRekishi** [comment]
[Account: Fix login with wrong username](https://trello.com/c/kWuqc5qx/205-account-fix-login-with-wrong-username)

File: routes/user.js [hunk 1]
```diff
@@ -7,6 +7,21 @@ const bcrypt = require("bcrypt");
...
+const authenticateToken = (req, res, next) => {
+  const authHeader = req.headers["authorization"];
+  const token = authHeader && authHeader.split(" ")[1];
+
+  if (token == null) {
+    return res.status(401).send("No token registered");
+  }
+
+  jwt.verify(token, process.env.ACCESS_TOKEN_SECRET, (err, user) => {
+    if (err) return res.status(403).send("The token is unvalid");
```
Comment (LordRekishi) [inline routes/user.js]:
```suggestion
    if (err) return res.status(403).send("The token is invalid");
```
Not a major thing, but a more common word is invalid

File: routes/user.js [hunk 2]
```diff
@@ -46,50 +76,32 @@ router.post("/user/login", async (req, res) => {
 
   rowData = [];
   let findUser = userDB.prepare("SELECT * FROM users WHERE username = ?");
+  console.log("here");
```
Comment (LordRekishi) [inline routes/user.js]:
```suggestion
```
Guessing you left this by mistake?

File: routes/user.js [hunk 3]
```diff
@@ -25,18 +40,33 @@ router.get("/user", authenticateToken, (req, res) => {
...
-    const hashedPassword = await bcrypt.hash(req.body.password, 10);
-    const user = { username: req.body.name, password: hashedPassword };
+  let findUser = userDB.prepare("SELECT * FROM users WHERE username = ?");
 
-    let insert = "INSERT INTO users (username, password) VALUES (?,?)";
-    userDB.run(insert, [user.username, user.password]);
+  //Change to get
+  findUser.each(
+    req.body.name,
+    (error, row) => {},
```
Comment (LordRekishi) [inline routes/user.js:48]: (error, row) => {}. This row is not used? Needed?

**Review by LordRekishi — CHANGES_REQUESTED** [review]
Functionality wise it works great! Made some comments

File: routes/user.js (hunk 3 above)
Comment (FelixJacobsen) [reply routes/user.js:48]: Yeah otherwise the rest of the code is used in the loop and it causes errors

**Review by LordRekishi — APPROVED** [review]

**Comment by LordRekishi** [comment]
Looks great now, just need to resolve the conflict and its ready to go

**Pull request merged**